source/php/Helper/QueryEvents.php

```diff
@@ -71,17 +73,48 @@ public static function getEventsByInterval($params, $page = 1)
         SELECT      *, $wpdb->posts.ID AS ID
         FROM        $wpdb->posts
         LEFT JOIN   $db_table ON ($wpdb->posts.ID = $db_table.event_id) ";
-        $query .= ($ageGroup) ? "LEFT JOIN $wpdb->postmeta age_from ON $wpdb->posts.ID = age_from.post_id " : '';
-        $query .= ($ageGroup) ? "LEFT JOIN $wpdb->postmeta age_to ON $wpdb->posts.ID = age_to.post_id " : '';
+
+		$query .= "LEFT JOIN $wpdb->postmeta AS mt1 ON ( $wpdb->posts.ID = mt1.post_id ) ";
+		$query .= "LEFT JOIN $wpdb->postmeta AS mt2 ON ( $wpdb->posts.ID = mt2.post_id AND mt2.meta_key = 'age_group_to' ) ";
+		$query .= "LEFT JOIN $wpdb->postmeta AS mt3 ON ( $wpdb->posts.ID = mt3.post_id AND mt3.meta_key = 'age_group_from' ) ";
+
         $query .= ($categories) ? "LEFT JOIN $wpdb->term_relationships term1 ON ($wpdb->posts.ID = term1.object_id) " : '';
         $query .= ($tags) ? "LEFT JOIN $wpdb->term_relationships term2 ON ($wpdb->posts.ID = term2.object_id) " : '';
         $query .= ($groups) ? "LEFT JOIN $wpdb->term_relationships term3 ON ($wpdb->posts.ID = term3.object_id) " : '';
         $query .= ($languageId) ? "LEFT JOIN $wpdb->term_relationships term4 ON ($wpdb->posts.ID = term4.object_id) " : '';
+
         $query .= "
         WHERE $wpdb->posts.post_type = %s
         AND $wpdb->posts.post_status = %s
         AND ($db_table.start_date BETWEEN %s AND %s OR $db_table.end_date BETWEEN %s AND %s) ";
 
+		if ( $ageMin ) {
+			$query .= "AND ( 
+				( 
+				  ( mt2.meta_key = 'age_group_from' AND mt2.meta_value <= $ageMin )
+				) 
+				OR 
```

Comment:
We need a third condition to match events where mt2.meta_key IS NULL but has age_group_to set (regardless of value, simply not IS NULL). Same thing in reverse for the $ageMax check on :104